state/fork/traversal.go

```diff
@@ -107,9 +107,11 @@ func unsafeTraverse(headers storage.Headers, block *flow.Header, visitor onVisit
 			return block, nil
 		}
 
-		block, err = headers.ByBlockID(block.ParentID)
+		parent, err := headers.ByBlockID(block.ParentID)
 		if err != nil {
-			return nil, fmt.Errorf("failed to revtrieve block header %x: %w", block.ParentID, err)
+			return nil, fmt.Errorf("failed to revtrieve block header %v (%x): %w", block.Height, block.ParentID, err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed to retrieve block header (id=%x height=%d): %w", block.ParentID, block.Height-1, err)
```
Suggesting to adjust the error message so that both variables refer to the same block (parent) and are labelled. Currently the height is for the child and the id is for the parent.